Automattic/wp-calypso
# Add domain management restrictions for Gravatar domains

## Proposed Changes

This PR updates domain settings and domain transfers pages to support domains used with Gravatar profiles.

For these domains, only these features are enabled:
- contact info update;
- external transfer out (and transfer lock);
- cancelation;
- renewal;
- enable/disable autorenew;

## Testing Instructions
Apply this PR locally (or use the Calypso link below) and set the `gravatar` flag (or force `is_gravatar_domain` property in the `Domain_Managament_Domain` class) on a **domain only site**.

Verify that the domain setting/transfer options are restricted (use screenshots below as a reference).

![gravatar-1](https://github.com/Automattic/wp-calypso/assets/2797601/bfb94f63-68d7-4527-8530-1abd9cb5b9f3)

![gravatar-2](https://github.com/Automattic/wp-calypso/assets/2797601/5276cb0e-f23d-4dd9-a681-9ac6b20ae27d)

## Pre-merge Checklist
